service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendControllerTest.java

```diff
@@ -0,0 +1,269 @@
+    public static final String ISSUE_CODE_BUSINESS_RULE = "business-rule";
+
+    private ObjectMapper objectMapper;
+
+    @Mock
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    @Mock
+    private TimestampService timestampService;
+
+    @Mock
+    private RandomIdGeneratorService randomIdGeneratorService;
+
+    @Mock
+    private TaskDispatcher taskDispatcher;
+
+    private EhrResendController ehrResendController;
+
+
+    @BeforeEach
+    void setUp() {
+        ObjectMapperBean objectMapperBean = new ObjectMapperBean();
+        objectMapper = objectMapperBean.objectMapper(new Jackson2ObjectMapperBuilder());
+        FhirParseService fhirParseService = new FhirParseService();
+        ehrResendController = new EhrResendController(ehrExtractStatusRepository,
+                                                      taskDispatcher,
+                                                      randomIdGeneratorService,
+                                                      timestampService,
+                                                      fhirParseService);
+    }
+
+    @Test
+    void When_AnEhrExtractHasFailed_Expect_GetGpcStructuredTaskScheduled() {
+
+        String ehrMessageRef = generateRandomUppercaseUUID();
+        var ehrExtractStatus = new EhrExtractStatus();
+
+        ehrExtractStatus.setConversationId(CONVERSATION_ID);
+        ehrExtractStatus.setEhrReceivedAcknowledgement(EhrExtractStatus.EhrReceivedAcknowledgement.builder()
+                                                           .conversationClosed(FIVE_DAYS_AGO)
+                                                           .errors(List.of(
+                                                               EhrExtractStatus.EhrReceivedAcknowledgement.ErrorDetails.builder()
+                                                                   .code(INCUMBENT_NACK_CODE)
+                                                                   .display(INCUMBENT_NACK_DISPLAY)
+                                                                   .build()))
+                                                           .messageRef(ehrMessageRef)
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+        ehrExtractStatus.setEhrExtractCorePending(EhrExtractStatus.EhrExtractCorePending.builder().build());
+        ehrExtractStatus.setEhrContinue(EhrExtractStatus.EhrContinue.builder().build());
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder().build());
+
+        doReturn(Optional.of(ehrExtractStatus)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
```

Comment:
I don't think we should be using [doReturn()](https://javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/Mockito.html#doReturn(java.lang.Object)) here, as it should on be used when:

> Use doReturn() in those rare occasions when you cannot use [when(Object)](https://javadoc.io/static/org.mockito/mockito-core/5.14.2/org/mockito/Mockito.html#when(T)).
Beware that [when(Object)](https://javadoc.io/static/org.mockito/mockito-core/5.14.2/org/mockito/Mockito.html#when(T)) is always recommended for stubbing because it is argument type-safe and more readable (especially when stubbing consecutive calls).